server/internal/instrument/prometheus.go

```diff
@@ -167,6 +169,31 @@ var (
 	)
 )
 
+var monitoredChannels = struct {
+	sync.Mutex
+	channels map[string]chan interface{}
+}{
+	channels: make(map[string]chan interface{}),
+}
+
+func startChannelLenMonitor() {
```

Comment:
this isn't ideal. we do not need nested for loops or mutex contention or a map storing things.

Reply:
i fixed this by pushing another commit on this branch